### DISCUSSION TITLE
Air Purifier Controls

### DISCUSSION THREAD

**Discussion by jakejcarroll** [Andre0512/hon#85] [Q&A]
Hi Andre,

Thanks so much for this integration!

I know you're still in beta for air purifier models, but I'm wondering if the capability exists to override the default options that exist through Hon & manually set the fan speed? Through the app you are only able to set the speed to auto, max or sleep which is disappointing, and I doubt this is something which can be overridden to manually set the fan speed...but worth the ask!

Thanks again :)

**Comment by Andre0512** [comment]
Hi, can you show me the data of your device? Then I can have a look if this is possible :slightly_smiling_face: 
I have seen that there might be devices that support this, but the device I have integrated so far does not offer this...


> **Data Archive**
> For further analysis, please add your appliance data archive (if available)
> Navigate to `Settings` -> `Device & Services` -> `Haier hOn` -> _your device_ and press the _Create Data Archive_ button. 
> Then open notifications to download the data zip archive.
> To attach the file:
> * GitHub Web: Use the "Attach files by dragging & dropping, selecting or pasting them." function
> * GitHub Mobile: Upload the zip archive as image

**Reply by jakejcarroll** [reply]
I've attached the device data :) Have a feeling the model which would support this is the 700 model & I have the 500, but hopefully I'm wrong! Thanks again 


[ap_137.zip](https://github.com/Andre0512/hon/files/11923643/ap_137.zip)

**Reply by Andre0512** [reply]
Hi, sorry this can't be overwritten by your appliance...
